plonk-core/src/lookup/preprocess.rs

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::commitment::HomomorphicCommitment;
+use crate::error::{to_pc_error, Error};
+use crate::lookup::{LookupTable, MultiSet};
+use ark_ff::PrimeField;
+use ark_poly::domain::EvaluationDomain;
+use ark_poly::polynomial::univariate::DensePolynomial;
+
+/// This table will be the preprocessed version of the precomputed table,
+/// T, with arity 4. This structure is passed to the proof alongside the
+/// table of witness values.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct PreprocessedLookupTable<F, PC>
```

Comment:
Basically because we only support 4 wire circuits. As I mentioned above we should introduce the flexibility in the table width and number of wires at the same time, but that is a significant change worth its own dedicated PR.